### ISSUE TITLE
scipy.signal.place_poles real pole ordering?

### ISSUE THREAD

**Issue by vmatos** [scipy/scipy#6692]
Is there a reason why I shouldn't be able to choose the pole value for the specific state on my MIMO system?
For instance in matlab:
```
p = [-1., -.5, -.4, -1.5];
L = place(A',C',p)'
% Matlab:
% L =
%     1.0000         0         0    1.0000
%   -42.7207    0.5000         0         0
%          0         0    0.4000         0
%    47.0335         0         0    1.5000
```

With scipy:
```
ff_fbk = scipy.signal.place_poles(A.T, C.T, p)
L_sp = ff_fbk.gain_matrix.T
# [[  1.5         -0.          -0.           1.        ]
#  [-42.72073069   1.          -0.          -0.        ]
#  [ -0.          -0.           0.5         -0.        ]
#  [ 47.03349017  -0.          -0.           0.4       ]]
```

I looked a bit into it, and found out that in `_order_complex_poles` (https://github.com/scipy/scipy/blob/master/scipy/signal/ltisys.py#L2548), real poles are also ordered.
Making this line:
```
    ordered_poles = poles[np.isreal(poles)]
```
Produces the same output as Matlab.

**Comment by ilayn** [comment]
Note that, unlike SISO systems, the solution for pole placement for MIMO systems is not unique but you should get the same closed loop eigenvalues for A - LC in both cases. I'm not sure what you mean by pole value for specific state.

**Comment by vmatos** [comment]
Take for instance my observer design. I want the observer error to decay faster in state x1 than x2. Therefore I choose to place p1 further from 0 than p2. Is that correct?

**Comment by ilayn** [comment]
The order of the numbers appearing in the vector and the states are not related. It is the resulting pole locations of the A-LC matters. You don't get to choose which mode is moved to where in MIMO problems. Like I mentioned above, the solution is not unique. Check your eigenvalues and compare it with matlab's.

**Comment by vmatos** [comment]
Yes, all this time I've been comparing every result with Matlab. Why would Matlab respect the order I give it for the poles, if it doesn't matter?

The eigenvalues are the same, but not the eigenvectors. Isn't the whole eigenstructure relevant for the shape of the response? http://users.isy.liu.se/en/rt/johans/files/articles/andry_etal_83.pdf

**Comment by ilayn** [comment]
As long as your system is observable, I don't see why it should matter. Example: 

    G = rss(5,2,3);
    p = -[1:5];
    eig(G.a - place(G.a',G.c',p)'*G.c)
    eig(G.a - place(G.a',G.c',fliplr(p))'*G.c)

both gives the same order of eigenvalues. For eigenvectors, most probably they are only different up to orthonormalization. But I'm not sure what matlab does by heart.

**Comment by vmatos** [comment]
From https://www.quora.com/Control-Theory-What-is-the-pole-placement-method/answer/Sam-Nazari-2?srid=ZWbO
>The eigenvalues of the system determine the stability (and to some extent bandwidth) of the closed loop system while the eigenvectors determine the degree to which each eigenvalue influences the response of each state variable. The main reason why one would utilize eigenstructure assignment is to have total control over the modal response of the system. 

Am I correctly understanding that this statement contradicts your previous statement?
> You don't get to choose which mode is moved to where in MIMO problems.

**Comment by ilayn** [comment]
No actually, that says that there is no guarantee that you will get the order in a prescribed manner. I wanted to give an example that even the eigenvalue order is the same, the resulting gain matrices are different which is the consequence of nonuniqueness of the solution. If you flip the eigenvectors then you can change the order of the eigenvalues and that's just going to be a permutation hence nothing will change. 

Because what is meant in that eigenstructure assignment in the link is that you basically end up with a prescribed set of eigenvalues. And a subspace spanned by those eigenvectors. If you have an observable system, that subspace is the whole R^n. Of course, you can choose any kind of independent vectors to span the same space. That's why eigenvector set can be different as long as they are linearly independent.

Because of this independence, you can actually look for different optimality conditions. You might look for the minimum gain matrix, or any other norm condition etc. that leads to a convex problem using the degrees of freedom the MIMO problem has.

**Comment by person142** [comment]
Wading into a bunch of jargon that I'm not familiar with, but IIRC all this comes down to: find a matrix `K` so that `A - BK` has a prescribed set of eigenvalues. The solution isn't unique, and currently SciPy and Matlab find different solutions.

So there's no bug in SciPy here--it only claims that `A - BK` will have the prescribed eigenvalues, and indeed it does. (Though the first sentence of the docstring for `place_poles` could probably be phrased more clearly.)

Matlab claims (https://se.mathworks.com/help/control/ref/place.html) their algorithm 

> uses the extra degrees of freedom to find a solution that minimizes the sensitivity of the closed-loop poles to perturbations in *A* or *B*.

So re

> Why would Matlab respect the order I give it for the poles, if it doesn't matter?

it doesn't seem like Matlab claims it will necessarily respect the order you give it for the poles. Though this does leave open the question of whether Matlab's claim is true and it is getting a "better" solution. (Though as @ilayn pointed out one could want optimality conditions other than "minimal sensitivity to perturbations".)

Does someone more qualified (@Eric89GXL @e-q @endolith) than me want to weigh in and save me from myself?
